src/hooks/useAuthenticated.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { USER_SELF, DEFAULT_AVATAR } from '@/components/constants/url';
+
+type Userdata = {
+  userName: string;
+  firstName: string;
+  profilePicture: string;
+};
+
+type HooksReturnType = {
+  userData: Userdata;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+};
+
+const useAuthenticated = (): HooksReturnType => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [userData, setUserData] = useState<Userdata>({
+    userName: '',
+    firstName: '',
+    profilePicture: DEFAULT_AVATAR
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      await fetch(USER_SELF, { credentials: 'include' })
+        .then((response) => {
+          if (!response.ok) {
+            setIsLoggedIn(false);
+            throw new Error(`${response.status} (${response.statusText})`);
+          }
+          return response.json();
+        })
+        .then((responseJson) => {
+          if (responseJson.incompleteUserDetails) {
+            window.open(
+              'https://my.realdevsquad.com/signup',
```

Comment:
Could we ensure we fetch link from constant

Reply:
Done